src/components/TagSet/TagSet.tsx

```diff
@@ -0,0 +1,101 @@
+import { As, chakra, Flex, useStyleConfig } from "@chakra-ui/react";
+import React, { forwardRef } from "react";
+import Tooltip from "../Tooltip/Tooltip";
+
+import TagSetExplore, {
+  TagSetExploreProps,
+  TagSetExploreDataProps,
+} from "./TagSetExplore";
+import TagSetFilter, {
+  TagSetFilterDataProps,
+  TagSetFilterProps,
+} from "./TagSetFilter";
+
+export interface BaseTagSetProps {
+  /** Additional class for the componen.t */
+  className?: string;
+}
+
+// We want either the "explore" or "filter" type props.
+export type TagSetTypeProps = TagSetFilterProps | TagSetExploreProps;
+// And here combine the special types with the base props.
+export type TagSetProps = BaseTagSetProps & TagSetTypeProps;
+
+// Type guard so we can make sure we have a "filter" `TagSet` variant.
+export function isFilterType(type: TagSetProps["type"]): type is "filter" {
+  return type === "filter";
+}
+
+/**
+ * This helper component wrapper renders a DS `Tooltip` component if the text is
+ * long or a React fragment. This assumes that the `label` prop is a rather
+ * simple single root JSX element, such as `<Link ...>....</Link>`.
+ */
+export const ToolTipWrapper = ({
+  label,
+  children,
+}: React.PropsWithChildren<{ label: any }>) => {
+  const labelText = typeof label === "string" ? label : label.props.children;
+  if (labelText.length > 17 && typeof labelText === "string") {
```

Comment:
How is 17 determined? Should this be made a constant to avoid magic numbers?

Reply:
It was determined kinda randomly but it's more-or-less the character length before the tag reaches 200px width. It IS a magic number and I will fix it.

Reply:
@EdwinGuzman Sorry, I missed this the first time around.  Did you consider using a method similar to this?  https://thewebdev.info/2021/04/17/how-to-check-for-text-overflow-ellipsis-in-an-html-element/

Reply:
@EdwinGuzman Let’s stay with the character count solution, but maybe bump it up to “20" so it doesn’t seem to random.  Three more chars won’t increase the width so much that is feels too big.